admin/spam/views.py

```diff
@@ -1,47 +1,106 @@
+from django.core.urlresolvers import reverse
+from django.views.defaults import page_not_found
 
 from modularodm import Q
 from website.project.model import Comment
 
 from .serializers import serialize_comment
+from .forms import ConfirmForm
 
 
-def get_spam_list():
-    query = (
-        Q('reports', 'ne', {}) &
-        Q('reports', 'ne', None)
-    )
-    return Comment.find(query).sort('date_created')
+class SpamList(ListView):
+    template_name = 'spam/spam.html'
+    paginate_by = 10
+    paginate_orphans = 1
+    ordering = 'date_created'
+    context_object_name = 'Spam'
 
+    def __init__(self):
+        self.status = str(Comment.FLAGGED)
+        super(SpamList, self).__init__()
 
-@login_required
-def spam_list(request):
-    paginator = Paginator(get_spam_list(), 10)
-
-    page_number = request.GET.get('page', 1)
-    try:
-        page = paginator.page(page_number)
-    except PageNotAnInteger:
-        page = paginator.page(1)
-    except EmptyPage:
-        page = paginator.page(paginator.num_pages)
-    context = {
-        'spam': map(serialize_comment, page),
-        'page': page,
-        'page_number': page_number,
-    }
-    return render(request, 'spam/spam.html', context)
+    def get_queryset(self):
+        self.status = self.request.GET.get('status', u'1')
+        query = (
+            Q('reports', 'ne', {}) &
+            Q('reports', 'ne', None) &
+            Q('spam_status', 'eq', int(self.status))
+        )
+        return Comment.find(query).sort(self.ordering)
 
+    def get_context_data(self, **kwargs):
+        queryset = kwargs.pop('object_list', self.object_list)
+        page_size = self.get_paginate_by(queryset)
+        paginator, page, queryset, is_paginated = self.paginate_queryset(
+            queryset, page_size)
+        return {
+            'spam': map(serialize_comment, queryset),
+            'page': page,
+            'status': self.status,
+            'page_number': page.number
+        }
 
-@login_required
-def spam_detail(request, spam_id):
-    context = {
-        'comment': serialize_comment(Comment.load(spam_id)),
-        'page_number': request.GET.get('page', 1),
-    }
-    return render(request, 'spam/comment.html', context)
+
+class SpamDetail(FormView):
+    form_class = ConfirmForm
+    template_name = 'spam/comment.html'
+    spam_id = None
+    page = 1
+
+    def __init__(self):
+        self.item = None
+        super(SpamDetail, self).__init__()
+
+    @method_decorator(login_required)
+    def get(self, request, *args, **kwargs):
+        try:
+            context = self.get_context_data(**kwargs)
+        except AttributeError:
+            return page_not_found(request)  # TODO: 1.9 update to have exception with node/user 404.html will be added
+        self.page = request.GET.get('page', 1)
+        context['page_number'] = self.page
+        context['form'] = self.get_form()
+        return self.render_to_response(context)
```

Comment:
Again, unless you're trying to avoid some specific behavior it's good to return super() from CBV methods.